sdk/storage/Azure.Storage.Blobs/src/BlobAudience.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+using Azure.Core;
+
+namespace Azure.Storage.Blobs
+{
+    /// <summary>
+    /// Audiences available for Blobs
+    /// </summary>
+    public readonly partial struct BlobAudience : IEquatable<BlobAudience>
+    {
+        private readonly string _audience;
+
+        /// <summary>
+        /// Intializes new instance of <see cref="BlobAudience"/>.
+        /// </summary>
+        /// <param name="audience">
+        /// The Azure Active Directory audience to use when forming authorization scopes.
+        /// For the Language service, this value corresponds to a URL that identifies the Azure cloud where the resource is located.
+        /// For more information: <see href="https://learn.microsoft.com/en-us/azure/storage/blobs/authorize-access-azure-active-directory" />.
+        /// </param>
+        /// <remarks>Please use one of the static constant members over creating a custom value unless you have specific scenario for doing so.</remarks>
+        public BlobAudience(string audience)
+        {
+            Argument.AssertNotNullOrEmpty(audience, nameof(audience));
+            _audience = audience;
+        }
+
+        private const string _publicAudience = "https://storage.azure.com/.default";
+
+        /// <summary>
+        /// Default Audience. Use to acquire a token for authorizing requests to any Azure Storage account
+        ///
+        /// Resource ID: &quot;https://storage.azure.com/.default &quot;.
+        ///
+        /// If no audience is specified, this is the default value.
+        /// </summary>
+        public static BlobAudience PublicAudience { get; } = new(_publicAudience);
+
+        /// <summary>
+        /// The service endpoint for a given storage account.
+        /// Use this method to acquire a token for authorizing requests to that specific Azure Storage account and service only.
+        /// </summary>
+        /// <param name="storageAccountName"></param>
+        /// <returns></returns>
+        public static BlobAudience BlobServiceAccountAudience(string storageAccountName) => new($"https://{storageAccountName}.blob.core.windows.net/.default");
```

Comment:
Not sure if this something we wanted officially but I could see this being a common scenario for customers.

See
https://learn.microsoft.com/en-us/azure/storage/blobs/authorize-access-azure-active-directory#microsoft-authentication-library-msal

Reply:
Personally, I like this convenience method as it helps users to easily scope tokens to their specific account. That said I would defer to @tg-msft with regards to the naming and shape of this static factory method.

Reply:
This is a reasonable helper to add, but we should make it look like a method (i.e., include a verb in the name) and add something to hint that it's for public Azure.  Maybe `GetBlobPublicAudience`?

Reply:
Done